src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -183,6 +192,34 @@ internal override CommonAnonymousTypeManager CommonAnonymousTypeManager
         /// </summary>
         internal bool IsPeVerifyCompatEnabled => LanguageVersion < LanguageVersion.CSharp7_2 || Feature("peverify-compat") != null;
 
+        internal bool RunNullableWalker
+        {
+            get
+            {
+                if (!_lazyRunNullableWalker.HasValue())
+                {
+                    if (Options.NullableContextOptions != NullableContextOptions.Disable)
```

Comment:
Consider changing to `Options.NullableContextOptions == NullableContextOptions.Enable || Options.NullableContextOptions == NullableContextOptions.Warnings`

since the checks we do for the "annotations" context take place in type binding, not in nullable analysis (iirc).

Reply:
There are a few places which do this check, it might even be time for an extension method `bool NullableContextOptions.WarningsEnabled()`. Your choice whether to do that in this PR though.

Reply:
No, that was what I had thought about doing in the first place, but discussions with Julien disuaded me. If annotations are on you'll want to see IDE features light up with the information, so we'll need to run the analysis.